src/pages/config/index.js

```diff
@@ -113,29 +128,112 @@ export class Config extends Component {
     this.setState({ [e.target.name]: e.target.value })
   }
 
-  // config subjkt
+  name_exists = async () => {
+    if (!this.state.subjkt) return false
+
+    const { errors, data } = await fetchGraphQL(
+      query_name_exist,
+      'nameExists',
+      {
+        name: this.state.subjkt,
+      }
+    )
+    if (errors) {
+      console.error(errors)
+      return false
+    }
+    if (data.holder.length == 0) return false
+
+    const holder = data.holder[0]
+
+    if (holder.address == this.state.address) return false
 
+    console.error(`name exists and is registered to ${holder.address}`)
+
+    this.context.setFeedback({
+      visible: true,
+      message: `The provided name is already registered by ${holder.address}`,
+      progress: false,
+      confirm: true,
+      confirmCallback: () => {
+        this.context.setFeedback({ visible: false })
+      },
+    })
+
+    return true
+  }
+
+  // upload to profile pic + subjkt meta to IPFS & call the SUBJKT registry
   subjkt_config = async () => {
+    if (await this.name_exists()) {
+      return
+    }
+
+    this.context.setFeedback({
+      visible: true,
+      message: 'uploading SUBJKT',
+      progress: true,
+      confirm: false,
+    })
+
     if (this.state.selectedFile) {
       const [file] = this.state.selectedFile
 
+      this.context.setFeedback({
+        message: 'uploading indenticon',
+      })
+
       const buffer = Buffer.from(await file.arrayBuffer())
-      console.log(buffer)
-      this.setState({ identicon: 'ipfs://' + (await ipfs.add(buffer)).path })
+      const picture_cid = await uploadFileToIPFSProxy({
```

Comment:
i'm not sure, but can't `file` just be passed directly to `uploadFileToIPFSProxy`?

Reply:
Probably!

This was a bit of the trouble in this PR, that's why I roughly typed it `@typedef { {path: string?, blob: Blob} } FileHolder`, extracting it from the old code and it's usage mainly to make sense of the code. I would happily clean it further in a separate branch/PR later

Reply:
ok, yeah, no need that we need to block merging this pr because of that.